### PR TITLE
ref #773: Add missing files

### PR DESCRIPTION
**Pull request by UlrichKu**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#773
| License       | MIT

### CODE REVIEW

**Comment by pixeljunkie** [comment]
We should not add files with .class.php as names, even if this was their original filename. There is no need for this additional suffix anymore. So i would prefer to have the suffix removed.

**Comment by j6s** [comment]
Just so you are aware: CI checks are not yet in `master`, but a previous run of Psalm with the incorrect base branch showned a couple of missing type declarations: https://github.com/chameleon-system/chameleon-base/actions/runs/2274339156

File: src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    }
+
+    /**
+     * in some cases (probably due to php bug #55737) php is not able to perform a LOAD DATA
+     * LOCAL INFILE using a non-root mysql user. In this case we'll fall back to an import
+     * call via shell. Be sure the user is allowed to perform a LOAD DATA LOCAL INFILE command.
+     *
+     * @param string $sQuery
+     *
+     * @return mixed (0 on success)
```
Comment (j6s) [inline src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php:144]: Based on the implementation, the return type should probably be `int|false`. The important distinction being that `false` and non-0 denotes an error while 0 denotes success

File: src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    {
+        if (null !== $this->dbConnection) {
+            $this->dbConnection->close();
+        }
+    }
+
+    /**
+     * returns true if the init was ok - else false.
+     *
+     * @param $sTable
```
Comment (j6s) [inline src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php:70]:
```suggestion
     * @param string $sTable
```

File: src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_SimpleSql.class.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+     *
+     * @var array
+     */
+    private $aFields = array();
+
+    /**
+     * returns true if the init was ok - else false.
+     *
+     * @param $sTable string the target table name
+     * @param $aFields array list of fields in target table
```
Comment (j6s) [inline src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_SimpleSql.class.php:37]:
```suggestion
     * @param string[] $aFields list of fields in target table
```

File: src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_SimpleSql.class.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+     * list of fields in target table.
+     *
+     * @var array
+     */
+    private $aFields = array();
+
+    /**
+     * returns true if the init was ok - else false.
+     *
+     * @param $sTable string the target table name
```
Comment (j6s) [inline src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_SimpleSql.class.php:36]:
```suggestion
     * @param string $sTable the target table name
```

File: src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_SimpleSql.class.php [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+     * the target table name.
+     *
+     * @var null|string
+     */
+    private $sTableName = null;
+
+    /**
+     * list of fields in target table.
+     *
+     * @var array
```
Comment (j6s) [inline src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_SimpleSql.class.php:29]:
```suggestion
     * @var string[]
```

File: src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+     * @param $aFields
```
Comment (j6s) [inline src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php:71]:
```suggestion
     * @param string[] $aFields
```

File: src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+class TPkgCmsBulkSql_LoadDataInfile implements IPkgCmsBulkSql
+{
+    private $sTableName = null;
```
Comment (j6s) [inline src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php:14]:
```suggestion
    /**
     * @var string
     */
    private $sTableName = null;
```

File: src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+    private $aFields = array();
```
Comment (j6s) [inline src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php:15]:
```suggestion
    /**
     * @var string[]
     */
    private $aFields = array();
```

File: src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+            }
+        } else {
+            trigger_error("Can't open file for writing: ".$sFileName);
+            $iRetVal = false;
+        }
+
+        return $iRetVal;
+    }
+
+    protected function EscapeData($aData)
```
Comment (j6s) [inline src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php:172]:
```suggestion
    /**
     * @param array<string, string> $aData
     * @return array<string, string>
     */
    protected function EscapeData($aData)
```

File: src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+            $this->rFile = fopen($this->sFileName, 'wb');
+        }
+
+        return false !== $this->rFile;
+    }
+
+    /**
+     * return true if the data was writ ten to the target file.
+     *
+     * @param $aData
```
Comment (j6s) [inline src/Csv2SqlBundle/objects/BulkSql/TPkgCmsBulkSql_LoadDataInfile.class.php:92]:
```suggestion
     * @param array<string, string> $aData
```

File: src/Csv2SqlBundle/objects/BulkSql/IPkgCmsBulkSql.class.php [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+interface IPkgCmsBulkSql
+{
+    public function Initialize($sTable, $aFields);
```
Comment (j6s) [inline src/Csv2SqlBundle/objects/BulkSql/IPkgCmsBulkSql.class.php:14]:
```suggestion
    /**
     * @param string $sTable
     * @param string[] $aFields
     * @return bool
     */
    public function Initialize($sTable, $aFields);
```

**Review by j6s — COMMENTED** [review]
I tried to deduce as much type information as possible from the diff in the PR. Take my suggestions with a grain of salt as I did not check to see how the methods are being used outside of the added files themselves

**Pull request merged**